ros__parameters:
  # subscribers

  camera_info_topic: /CAM_FRONT/camera_info
  lidar_topic: /LIDAR_TOP
  detections_topic: /CAM_FRONT/detections

  # publishers
  filtered_lidar_topic: /filtered_lidar
  cluster_centroid_topic: /cluster_centroid
  bounding_box_topic: /bounding_box

  # lidar frames
  lidar_top_frame: LIDAR_TOP

  #camera detections
  cam_front_detections: /CAM_FRONT/detections
  cam_front_right_detections: /CAM_FRONT_RIGHT/detections
  cam_front_left_detections: /CAM_FRONT_LEFT/detections

  # filtering parameters
  ransac_params:
    distance_threshold: 0.5
    # maximum allowed distance a point can be from the estimated plane to still be considered an inlier
    max_iterations: 1000
    # the number of times ransac randomly samples a point to estimate a plane, higher number means more accurate plane estimation but heavier computation

  euclid_params:
    cluster_tolerance: 0.5
    #This defines the maximum distance between two points for them to be considered part of the same cluster
    min_cluster_size: 30 # small values create more noise 
    max_cluster_size: 800 # big values slightly slow down performance

  density_filter_params:
  # filter out clusters that are too sparse and too far away, on a scale of 0 to 1
    density_weight: 0.4 # increase to filter out more sparse clusters, least important
    size_weight: 0.6 # increase to filter out more large clusters, most important
    distance_weight: 0.5 # increase to filter out more far away clusters, moderately important
    score_threshold: 0.7 # threshold score for a cluster to be considered as a valid cluster, increase to keep more clusters

  merge_threshold: 1.2
  # the threshold distance between two clusters centroids for them to be merged into one cluster

  object_detection_confidence: 0.55


    